thalerj/openbmctool.py

```diff
@@ -0,0 +1,1490 @@
+            outputText += row + "\n"
+        return outputText
+    else:
+        return res.text + occres.text
+"""
+     prints out the bmc alerts
+       
+     @param host: string, the hostname or IP address of the bmc
+     @param args: contains additional arguments used by the sel sub command
+     @param session: the active session to use
+     @param args.json: boolean, if this flag is set to true, the output will be provided in json format for programmatic consumption 
+"""     
+def sel(host, args, session):
+
+    url="https://"+host+"/xyz/openbmc_project/logging/entry/enumerate"
+    httpHeader = {'Content-Type':'application/json'}
+    #print(url)
+    res = session.get(url, headers=httpHeader, verify=False, timeout=60)
+    return res.text
+
+
+"""
+     converts a boolean value to a human readable string value
+       
+     @param value: boolean, the value to convert
+     @return: A string of "Yes" or "No"
+""" 
+def boolToString(value):
+    if(value):
+        return "Yes"
+    else:
+        return "No"
+    
+"""
+     parses the esel data and gets predetermined search terms
+       
+     @param eselRAW: string, the raw esel string from the bmc
+     @return: A dictionary containing the quick snapshot data unless args.fullEsel is listed then a full PEL log is returned
+"""    
+def parseESEL(args, eselRAW):
+    eselParts = {}
+    esel_bin = binascii.unhexlify(''.join(eselRAW.split()[16:]))
+    #search terms contains the search term as the key and the return dictionary key as it's value
+    searchTerms = { 'Signature Description':'signatureDescription', 'devdesc':'devdesc',
+                    'Callout type': 'calloutType', 'Procedure':'procedure'}
+    
+    with open('/tmp/esel.bin', 'wb') as f:
+        f.write(esel_bin)
+    errlPath = ""
+    #use the right errl file for the machine architecture
+    arch = platform.machine()
+    if(arch =='x86_64' or arch =='AMD64'):
+        if os.path.exists('/opt/ibm/ras/bin/x86_64/errl'):
+            errlPath = '/opt/ibm/ras/bin/x86_64/errl'
+        elif os.path.exists('errl/x86_64/errl'):
+            errlPath = 'errl/x86_64/errl'
+        else:
+            errlPath = 'x86_64/errl'
+    elif (platform.machine()=='ppc64le'):
+        if os.path.exists('/opt/ibm/ras/bin/ppc64le/errl'):
+            errlPath = '/opt/ibm/ras/bin/ppc64le/errl'
+        elif os.path.exists('errl/ppc64le/errl'):
+            errlPath = 'errl/ppc64le/errl'
+        else:
+            errlPath = 'ppc64le/errl'
+    else:
+        print("machine architecture not supported for parsing eSELs")
+        return eselParts
+    
+    
+    
+    if(os.path.exists(errlPath)):
+        output= subprocess.check_output([errlPath, '-d', '--file=/tmp/esel.bin']).decode('utf-8')
+#         output = proc.communicate()[0]
+        lines = output.split('\n')
+        
+        if(hasattr(args, 'fullEsel')):
+            return output
+        
+        for i in range(0, len(lines)):
```

Comment:
This looks like it could do with some work, but we can iterate on it after its merged and we include some test cases